### PR TITLE
[3.4.2] DTLS retransmission timeout

### PR DESCRIPTION
**Pull request by nickAS21**
## Pull Request description

https://github.com/eclipse/leshan/issues/1002

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] PR name contains fix version. For example, "[3.3.4] Hotfix of some UI component" or "[3.4] New Super Feature".
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Required for internal contributors only.

## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.
- [x] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.

### CODE REVIEW

**Review by smatvienko-tb — APPROVED** [review]
Looks good to me

**Comment by smatvienko-tb** [comment]
PE: https://github.com/thingsboard/thingsboard-pe/pull/1136

File: application/src/main/resources/thingsboard.yml [hunk 1]
```diff
@@ -710,6 +710,8 @@ transport:
     dtls:
       # Enable/disable DTLS 1.2 support
       enabled: "${COAP_DTLS_ENABLED:false}"
+     # RFC7925_RETRANSMISSION_TIMEOUT_IN_MILLISECONDS = 9000
+      retransmission_timeout: "${COAP_DTLS_RETRANSMISSION_TIMEOUT_IN_MILLISECONDS:9000}"
```
Comment (ashvayka) [inline application/src/main/resources/thingsboard.yml:714]: Please take a look how we name parameters that ar configured in milliseconds. For example:
```
stats_print_interval_ms: "${SQL_ATTRIBUTES_BATCH_STATS_PRINT_MS:10000}"
```

There are 77 parameters which env variable names end with "_MS" and your "_MILLISECONDS" is unique. Rename the parameter.

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/config/LwM2MTransportServerConfig.java [hunk 2]
```diff
@@ -37,6 +37,10 @@
 @ConfigurationProperties(prefix = "transport.lwm2m")
 public class LwM2MTransportServerConfig implements LwM2MSecureServerConfig {
 
+    @Getter
+    @Value("${transport.coap.dtls.retransmission_timeout}")
```
Comment (ashvayka) [inline common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/config/LwM2MTransportServerConfig.java:41]: Why "coap" and not "lwm2m"?

File: transport/lwm2m/src/main/resources/tb-lwm2m-transport.yml [hunk 3]
```diff
@@ -103,6 +103,10 @@ transport:
     enabled: "${TB_TRANSPORT_LOG_ENABLED:true}"
     max_length: "${TB_TRANSPORT_LOG_MAX_LENGTH:1024}"
   # Local LwM2M transport parameters
+  coap:
```
Comment (ashvayka) [inline transport/lwm2m/src/main/resources/tb-lwm2m-transport.yml:106]: This is wrong.

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/config/LwM2MTransportServerConfig.java (hunk 2 above)
Comment (ashvayka) [reply common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/config/LwM2MTransportServerConfig.java:41]: Please replace with:

```
@Value("${transport.lwm2m.dtls.retransmission_timeout:9000}")
```

File: common/coap-server/src/main/java/org/thingsboard/server/coapserver/TbCoapDtlsSettings.java [hunk 4]
```diff
@@ -51,6 +56,9 @@ public class TbCoapDtlsSettings {
     @Value("${transport.coap.dtls.bind_port}")
     private Integer port;
 
+    @Value("${transport.coap.dtls.retransmission_timeout}")
```
Comment (ashvayka) [inline common/coap-server/src/main/java/org/thingsboard/server/coapserver/TbCoapDtlsSettings.java:59]: add default value:

```
@Value("${transport.coap.dtls.retransmission_timeout:9000}")
```

Comment (ashvayka) [reply common/coap-server/src/main/java/org/thingsboard/server/coapserver/TbCoapDtlsSettings.java:59]: This is required for backward-compatibility with old yml files.

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/config/LwM2MTransportServerConfig.java (hunk 2 above)
Comment (nickAS21) [reply common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/config/LwM2MTransportServerConfig.java:41]: Ok

File: transport/lwm2m/src/main/resources/tb-lwm2m-transport.yml (hunk 3 above)
Comment (nickAS21) [reply transport/lwm2m/src/main/resources/tb-lwm2m-transport.yml:106]: Ok

File: common/coap-server/src/main/java/org/thingsboard/server/coapserver/TbCoapDtlsSettings.java (hunk 4 above)
Comment (nickAS21) [reply common/coap-server/src/main/java/org/thingsboard/server/coapserver/TbCoapDtlsSettings.java:59]: OK

**Pull request merged**